Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -182,210 +173,468 @@ internal void OnClientChangeScene(string scenePath, SceneOperation sceneOperatio
+
+            ServerStartedSceneChange?.Invoke(scene.path, sceneOperation);
 
             if (!Server.LocalClientActive)
-                ApplyOperationAsync(scenePath, sceneOperation).Forget();
+                UnLoadSceneAsync(scene, sceneOperation).Forget();
 
             // notify all clients about the new scene
-            Server.SendToAll(new SceneMessage { scenePath = scenePath, sceneOperation = sceneOperation });
+            if (players.Length > 0)
+                NetworkServer.SendToMany(players,
+                    new SceneMessage { MainActivateScene = scene.path, SceneOperation = sceneOperation });
+        }
+
+        /// <summary>
+        ///     Allows server to fully load in a new scene and override current active scene.
+        /// </summary>
+        /// <param name="scenePath">The full path to the scene file or the name of the scene.</param>
+        public void ServerLoadSceneNormal(string scenePath)
+        {
+            if(!Server && !Server.Active) return;
+
+            var players = new INetworkPlayer[Server.Players.Count];
+            Server.Players.CopyTo(players);
+
+            ServerSceneLoading(scenePath, players, true);
+        }
+
+        /// <summary>
+        ///     Allows server to fully load in another scene on top of current active scene.
+        /// </summary>
+        /// <param name="scenePath">The full path to the scene file or the name of the scene.</param>
+        /// <param name="players">List of player's that are receiving the new scene load.</param>
+        /// <param name="shouldClientLoadNormally">Should the clients load this additively too or load it full normal scene change.</param>
+        public void ServerLoadSceneAdditively(string scenePath, INetworkPlayer[] players, bool shouldClientLoadNormally = false)
+        {
+            if (!Server && !Server.Active) return;
+
+            ServerSceneLoading(scenePath, players, shouldClientLoadNormally, SceneOperation.LoadAdditive);
         }
 
         /// <summary>
-        /// Called from ChangeServerScene immediately before NetworkSceneManager's LoadSceneAsync is executed
-        /// <para>This allows server to do work / cleanup / prep before the scene changes.</para>
+        ///     Allows server to fully unload a scene additively.
         /// </summary>
-        /// <param name="scenePath">Path of the scene that's about to be loaded</param>
-        internal void OnServerChangeScene(string scenePath, SceneOperation operation)
+        /// <param name="scene">The scene handle which we want to unload additively.</param>
+        /// <param name="players">List of player's that are receiving the new scene unload.</param>
+        /// <param name="shouldClientUnloadNormally">Should the clients unload this additively too or unload it full normal scene change.</param>
+        public void ServerUnloadSceneAdditively(Scene scene, INetworkPlayer[] players, bool shouldClientUnloadNormally = false)
```

Comment:
shouldClientUnloadNormally is unused

Reply:
It is unused for now because it is meant for end users to use. This was designed so that if users want to unload a scene additive and stay in a specific scene still or to fully unload the scene